contracts/AelinPool.sol

```diff
@@ -385,6 +406,57 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
         emit PurchasePoolToken(msg.sender, purchaseTokenAmount);
     }
 
+    /**
+     * @dev allows anyone to become a purchaser with a qualified erc721
+     * nft in the pool depending on the scenarios
+     *
+     * Scenarios:
+     * 1. each wallet holding a qualified NFT to deposit an unlimited amount of purchase tokens
+     * 2. certain amount of purchase tokens per wallet regardless of the number of qualified NFTs held
+     * 3. certain amount of Investment tokens per qualified NFT held
+     */
+    function purchasePoolScenario1and2for721(address _collectionAddress, uint256 _purchaseTokenAmount) public {
+        require(hasNftList, "pool does not have an NFT list");
+        require(block.timestamp < purchaseExpiry, "not in purchase window");
+        require(IERC721(_collectionAddress).balanceOf(msg.sender) > 0, "should own atleast 1 nft");
+
+        NftData memory tmpNftData = nftCollectionDetails[_collectionAddress];
+        require(tmpNftData.collectionAddress == _collectionAddress, "collection not in the pool");
+
+        if(tmpNftData.purchaseAmount > 0) {
```

Comment:
you might need to update your Default Formatter in vscode to be prettier to get the right format here on save